executor/src/witgen/affine_expression.rs

```diff
@@ -382,6 +384,12 @@ where
             offset &= !mask;
         }
 
+        if covered_bits > T::modulus() {
+            return Ok(EvalValue::incomplete(
+                IncompleteCause::OverflowingBitConstraints,
+            ));
+        }
```

Comment:
This is the main fix.

Reply:
Wait, why not `>= T::modulus()`? It probably does not matter since modulus + 1 is never a power of two

Reply:
True though!